src/Terminal/Terminal.ts

```diff
@@ -512,8 +512,11 @@ export class Terminal {
     this.contractOpen = true;
     const promptResult = await contract.prompt();
 
-    //Check if the contract still exists by the time the promise is fulfilled
-    if (server.getContract(contractPath) == null) {
+    // Get a new copy of the server, in case it changed while the prompt was open
+    const postPromptServer = Player.getCurrentServer();
```

Comment:
The player can legitimately have changed servers in the interim. It's not possible for the *actual* player to do it, since they're stuck in a modal, but scripts can.

You'll need to re-lookup the server by name, instead.

Reply:
Honestly don't know how this got in there. I specifically looked up `GetServer` to fetch by the host name. :laughing: